From 70dd5b37bceed3bddfd9bccde82af00284826bb8 Mon Sep 17 00:00:00 2001
From: "Ch1llb0/skin.osmc" <36623079+Ch1llb0@users.noreply.github.com>
Date: Wed, 20 Mar 2019 03:20:23 +0100
Subject: [PATCH] Skinshortcuts improvements

script-skinshortcuts.xml:
- adjust label of submenu edit option
- add option to change widget labels

Variables.xml:
- change skinshortcuts-window variable to show better window titles

addon.xml:
- update changelog

Changelog.md:
- update changelog
---
 16x9/Variables.xml            |  4 ++--
 16x9/script-skinshortcuts.xml | 12 +++++++++++-
 Changelog.md                  |  8 ++++++++
 addon.xml                     |  2 +-
 4 files changed, 22 insertions(+), 4 deletions(-)

diff --git a/16x9/Variables.xml b/16x9/Variables.xml
index 8fe878c..f5e9a0a 100755
--- a/16x9/Variables.xml
+++ b/16x9/Variables.xml
@@ -1137,8 +1137,8 @@
 
 	<!-- Variables for script-skinshortcuts.xml -->
 	<variable name="skinshortcuts-window">
-		<value condition="String.EndsWith(Window.Property(groupname),.1)">Customise Hub Widgets</value>
-		<value condition="!String.IsEqual(Window.Property(groupname),mainmenu)">Customise Hub Menu</value>
+		<value condition="String.EndsWith(Window.Property(groupname),.1)">$LOCALIZE[31095]</value>
+		<value condition="!String.IsEqual(Window.Property(groupname),mainmenu)">$ADDON[script.skinshortcuts 32072]</value>
 		<value>$ADDON[script.skinshortcuts 32071]</value>
 	</variable>
 
diff --git a/16x9/script-skinshortcuts.xml b/16x9/script-skinshortcuts.xml
index c5ddde2..469c49c 100755
--- a/16x9/script-skinshortcuts.xml
+++ b/16x9/script-skinshortcuts.xml
@@ -171,7 +171,7 @@
 				<control type="button" id="405">
 					<width>1220</width>
 					<height>60</height>
-					<label>$LOCALIZE[31071]</label>
+					<label>$ADDON[script.skinshortcuts 32072]</label>
 					<visible>String.IsEqual(Window.Property(groupname),mainmenu)</visible>
 				</control>
 
@@ -192,6 +192,15 @@
 					<visible>String.EndsWith(Window.Property(groupname),.1)</visible>
 				</control>
 
+				<!-- Change label-->
+				<control type="button" id="604">
+					<width>1220</width>
+					<height>40</height>
+					<label>  -  $ADDON[script.skinshortcuts 32025]</label>
+                    <onclick>RunScript(script.skin.helper.service,action=setskinshortcutsproperty,header=$ADDON[script.skinshortcuts 32025],property=widgetName)</onclick>
+                    <visible>String.EndsWith(Window.Property(groupname),.1)</visible>
+				</control>
+				
 				<!-- Change size option -->
 				<control type="button" id="601">
 					<width>1220</width>
@@ -213,6 +222,7 @@
 					<disabledcolor>$VAR[TextColor4]</disabledcolor>
 					<visible>String.EndsWith(Window.Property(groupname),.1)</visible>
 				</control>
+				
 				<!-- Fallback artwork -->
 				<control type="button" id="603">
 					<width>1220</width>
diff --git a/Changelog.md b/Changelog.md
index 589f131..53902dc 100644
--- a/Changelog.md
+++ b/Changelog.md
@@ -28,6 +28,7 @@ _Improved_
 - add wide list as music view
 - add music OSD album art size switch
 - adjust widget headings to always show and adjust animations to match widget animations
+- add option to change widget labels
 
 _Fixed_
 - show proper game widget title when not using skinshortcuts script
@@ -415,6 +416,8 @@ script-nextup-notification-StillWatchingInfo.xml:
 
 script-skin_helper_service-ColorPicker.xml:
 - replace colors by new color variables
+- fix ondown, onup, onright and onleft of lists
+- add new navigation indicator include
 
 script.skinshortcuts-static-xml:
 - hack game widget to show proper widget title
@@ -426,6 +429,9 @@ script.skinshortcuts-static-xml:
 
 script-skinshortcuts.xml:
 - replace colors by new color variables
+- fix ondown, onup, onright and onleft of lists
+- add new navigation indicator include
+- add option to change widget labels
 
 Settings.xml:
 - replace colors by new color variables
@@ -462,6 +468,8 @@ SkinSettings.xml:
 - add new adjust music OSD album art size option to advanced skin settings
 - change onclicks of customization and skin helper addon-related options for non-OSMC devices (where the scripts are not pre-installed) to open new dialogs
 - un-comment skinhelper widgets button
+- add page indicators to longer add-ons category
+- rework add-ons category
 
 SmartPlaylistEditor.xml:
 - replace colors by new color variables
diff --git a/addon.xml b/addon.xml
index b71ba68..c2cb536 100644
--- a/addon.xml
+++ b/addon.xml
@@ -14,7 +14,7 @@
 		<platform>all</platform>
 		<summary lang="en">The default skin for OSMC</summary>
 		<description lang="en">Original skin: Andy Morton[CR]Original design: Simon Brunton[CR]Skinner: Julian Michel</description>
-		<news>OSMC Kodi v18 update[CR][CR][B]New[/B][CR]- add new v18 subtitle settings OSD during fullscreen video playback[CR]- add new games section to match v18 requirements[CR]- add games section[CR]- add new resolution select button/dialog in video player[CR]- add player icon to now playing dialog[CR]- add new dependency button in addon info dialog to match v18 requirements[CR]- new color options (color sets, background gradients, adjustable opacity)[CR]- add PVR channel number input dialog[CR]- add PVR timeshift status dialog[CR]- add welcome dialog on non-OSMC devices[CR]- add director button to video info dialog[CR]- add new views (wide low, wall small, wall low, wall info, list info)[CR]- add new sub-menu indicator icon[CR]- add "Random TV shows" widget as new standard for TV shows home menu entry[CR]- add new dialog navigation indicators[CR][CR][B]Improved[/B][CR]- adjust syntax, values, labels and infobools to match v18 requirements[CR]- adjust PVR section to match v18 requirements[CR]- highlighting color now adjusts according to text color[CR]- streamline OSD animations[CR]- add missing adjustable plot fonts[CR]- let favourites dialog behave like a normal window[CR]- add file path and name to refresh button in video info dialog[CR]- add song/album year to music player[CR]- add wide list as music view[CR]- add music OSD album art size switch[CR]- adjust widget headings to always show and adjust animations to match widget animations[CR][CR][B]Fixed[/B][CR]- show proper game widget title when not using skinshortcuts script[CR]- highlighting is now more consistent[CR]- fix current position/time remaining and current time/end time for PVR playback[CR]- hide deprecated previous/next channel buttons in PVR playback OSD[CR]- fix background of subtitle settings window[CR]- fix layout of PVR playback dialogs[CR]- fix dialog list navigation</news>
+		<news>OSMC Kodi v18 update[CR][CR][B]New[/B][CR]- add new v18 subtitle settings OSD during fullscreen video playback[CR]- add new games section to match v18 requirements[CR]- add games section[CR]- add new resolution select button/dialog in video player[CR]- add player icon to now playing dialog[CR]- add new dependency button in addon info dialog to match v18 requirements[CR]- new color options (color sets, background gradients, adjustable opacity)[CR]- add PVR channel number input dialog[CR]- add PVR timeshift status dialog[CR]- add welcome dialog on non-OSMC devices[CR]- add director button to video info dialog[CR]- add new views (wide low, wall small, wall low, wall info, list info)[CR]- add new sub-menu indicator icon[CR]- add "Random TV shows" widget as new standard for TV shows home menu entry[CR]- add new dialog navigation indicators[CR][CR][B]Improved[/B][CR]- adjust syntax, values, labels and infobools to match v18 requirements[CR]- adjust PVR section to match v18 requirements[CR]- highlighting color now adjusts according to text color[CR]- streamline OSD animations[CR]- add missing adjustable plot fonts[CR]- let favourites dialog behave like a normal window[CR]- add file path and name to refresh button in video info dialog[CR]- add song/album year to music player[CR]- add wide list as music view[CR]- add music OSD album art size switch[CR]- adjust widget headings to always show and adjust animations to match widget animations[CR]- add option to change widget labels[CR][CR][B]Fixed[/B][CR]- show proper game widget title when not using skinshortcuts script[CR]- highlighting is now more consistent[CR]- fix current position/time remaining and current time/end time for PVR playback[CR]- hide deprecated previous/next channel buttons in PVR playback OSD[CR]- fix background of subtitle settings window[CR]- fix layout of PVR playback dialogs[CR]- fix dialog list navigation</news>
 	</extension>
 	<assets>
 		<icon>icon.png</icon>
